devito/operator.py

```diff
@@ -146,8 +146,6 @@ def __init__(self, nt, shape, dtype=np.float32, stencils=[],
                                      space_dims=self.space_dims,
                                      compiler=self.compiler, profile=profile,
                                      cache_blocking=cache_blocking, block_size=block_size)
-        self.propagator.time_loop_stencils_b = self.propagator.time_loop_stencils_b + getattr(self, "time_loop_stencils_pre", [])
-        self.propagator.time_loop_stencils_a = self.propagator.time_loop_stencils_a + getattr(self, "time_loop_stencils_post", [])
```

Comment:
Ok, so the key idea here is the **principle of graceful degradation**. By that I mean, that we essentially have two user APIs or abstraction layers, if you will:
* Operator is the high-level purely symbolic API and should be preferred as the more elegant way to define an equation. 
* Propagator is the slightly more low-level API, which requires more work on the input to get to the specified format and associated data objects, but it also allows certain operations to be "hard-coded" that we can't express as a clean sympy equation; for an example see FWI Gradient and Born. This essentially acts as a "backdoor" or "escape hatch" for when the higher-level abstraction does not provide the exact feature that is required.

Now, this principle is really important, since small problems that lie *just outside* the intended problem domain are usually the death of a DSL in practice. By removing the "shortcut" I actually wanted to highlight that things that do not conform with our current high-level abstraction, ie. the sparse point interpolation, can still be defined by the user, albeit in a slightly more "manual" way.